packages/core/src/actions/contracts/prepareWriteContract.ts

```diff
@@ -129,10 +127,7 @@ export async function prepareWriteContract<
   const minimizedAbi = minimizeContractInterface({
     abi: abi as Abi, // TODO: Remove cast and still support `Narrow<TAbi>`
     functionName,
-  }) as TAbi extends Abi
-    ? [ExtractAbiFunction<TAbi, TFunctionName>]
-    : readonly unknown[]
```

Comment:
By returning `readonly unknown[]`, we lost type info so can just return the same generic type `TAbi`